frontend/Movie_Pigeon/app/src/main/java/org/example/team_pigeon/movie_pigeon/MeFragment.java

```diff
@@ -53,9 +57,36 @@ public void onClick(View v) {
                 myTask.execute("bookmark");
             }
         });
+
+        logoutButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                Log.e(TAG, "Logout button pressed");
+                loggingOut();
+            }
+        });
         return view;
     }
 
+    private void loggingOut() {
+        Intent loadStartActivity = new Intent(getActivity(), StartActivity.class);
+        credential = new File(Environment.getExternalStorageDirectory() + "/MoviePigeon/Signin/credential");
+        if (credential.exists()) {
+            Log.e(TAG, "Deleting existing credentials");
+            credential.delete();
+        }
+
+        if (!credential.exists()) {
+            Log.e(TAG, "Starting sign in activity");
+            getActivity().startActivity(loadStartActivity);
+            Log.e(TAG, "Finishing current activity");
+            getActivity().finish();
+        } else {
```

Comment:
There is no IOException here in else since it only detects credential file not available but not trying to delete it. It is more like the user accidentally deleted the file while our app is running so when he tries to logout, the file is already gone and will be deleted twice. However, I will make a change to handle the problem of missing credential file.